src/Core/Managed/Shared/Extensibility/Implementation/Tracing/CoreEventSource.cs

```diff
@@ -312,6 +312,42 @@ public void FailedToSend(string msg, string appDomainName = "Incorrect")
                 this.nameProvider.Name);
         }
 
+        [Event(
+            25,
+            Message = "Failed to flush aggregated metrics. Exception: {0}.",
+            Level = EventLevel.Error)]
+        public void FailedToFlushMetricAggregators(string ex, string appDomainName = "Incorrect")
+        {
+            this.WriteEvent(
+                25,
+                ex ?? string.Empty,
+                this.nameProvider.Name);
+        }
+
+        [Event(
+            26,
+            Message = "Failed to snapshot aggregated metrics. Exception: {0}.",
+            Level = EventLevel.Error)]
+        public void FailedToSnapshotMetricAggregators(string ex, string appDomainName = "Incorrect")
+        {
+            this.WriteEvent(
+                26,
+                ex ?? string.Empty,
+                this.nameProvider.Name);
+        }
+
+        [Event(
+            27,
+            Message = "Failed to invoke metric processor. Exception: {0}.",
+            Level = EventLevel.Error)]
```

Comment:
ERROR IS CUSTOMER FACING. Get type of the processor. User-actionable. Suggest turning off processor if persists.